src/app.js

```diff
@@ -1 +1,40 @@
+/* eslint-disable no-console */
 'use strict';
+
+const fs = require('fs').promises;
+const path = require('path');
+
+async function copyFileWithValidation(sourcePath, destinationPath) {
+  try {
+    const fromPath = path.resolve(sourcePath);
+    const toPath = path.resolve(destinationPath);
+
+    if (fromPath === toPath) {
+      console.error('Source and destination paths are the same');
+
+      return;
+    }
+
+    await fs.access(fromPath);
```

Comment:
Consider adding error handling for the `fs.access` call. If the source file does not exist or is not accessible, it would be helpful to provide a specific error message to the user.